charts/virtual-kubelet/templates/deployment.yaml

```diff
@@ -30,15 +30,61 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
+{{- if eq .Values.provider "azure" }}
+{{- with .Values.providers.azure }}
+        - name: ACI_RESOURCE_GROUP
+          value: {{ required "aciResourceGroup is required" .aciResourceGroup }}
+        - name: ACI_REGION
+          value: {{ required "aciRegion is required" .aciRegion }}
+{{- if .targetAKS }}
+        - name: ACS_CREDENTIAL_LOCATION
+          value: /etc/acs/azure.json
+        - name: AZURE_TENANT_ID
+          value: {{ required "tenantId is required" .tenantId }}
+        - name: AZURE_SUBSCRIPTION_ID
+          value: {{ required "subscriptionId is required" .subscriptionId }}
+        - name: AZURE_CLIENT_ID
+          value: {{ required "clientId is required" .clientId }}
+        - name: AZURE_CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "vk.fullname" $ }}
```

Comment:
naive question, what does $ mean?

Reply:
`$` is always a reference to the root scope (we're currently in a `with` block to make referencing keys in `.Values.providers.azure` a bit less repetitive).

Reference at the bottom of this page: https://github.com/helm/helm/blob/01da56f956329bbfcef14ac164f9ab49081ad108/docs/chart_template_guide/variables.md

Reply:
Make sense. Thanks for the explanation.